nav2_util/src/robot_utils.cpp

```diff
@@ -66,4 +66,41 @@ bool getCurrentPose(
   return false;
 }
 
+bool getPoseInTargetFrame(
+  geometry_msgs::msg::PoseStamped & pose,
+  tf2_ros::Buffer & tf_buffer, const std::string target_frame,
+    const double transform_timeout)
+{
+  static rclcpp::Logger logger = rclcpp::get_logger("getPoseInTargetFrame");
+
+  try {
```

Comment:
This is super redundant to the other function, please refactor them or remove one such that we don't have identical calls to TF

Reply:
Ok, I will come up with something but the catch / RCLCPP_ERRORs will be common to both (and the logger too)